src/Plutarch/Extra/Time.hs

```diff
@@ -0,0 +1,159 @@
+{-# LANGUAGE OverloadedRecordDot #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE ViewPatterns #-}
+
+module Plutarch.Extra.Time (
+    PCurrentTime (..),
+    pcurrentTime,
+    currentTime,
+    passertCurrentTime,
+    pisWithinCurrentTime,
+    pisCurrentTimeWithin,
+) where
+
+import Control.Composition ((.*))
+import GHC.Records (HasField)
+import Plutarch.Api.V1 (
+    PExtended (PFinite),
+    PInterval (PInterval),
+    PLowerBound (PLowerBound),
+    PPOSIXTime,
+    PUpperBound (PUpperBound),
+ )
+import Plutarch.Api.V2 (PPOSIXTimeRange)
+import Plutarch.Extra.Applicative (pliftA2)
+import Plutarch.Extra.Field (pletAll, pletAllC)
+import Plutarch.Extra.Maybe (passertPJust, pjust, pnothing)
+import Plutarch.Extra.TermCont (pmatchC)
+
+{- | Represent a fully bounded time range.
+
+     Note: 'PCurrentTime' doesn't need a Haskell-level equivalent because it
+     is only used in scripts, and does not go in datums. It is also scott-encoded
+     which is more efficient in usage.
+
+     @since 3.3.0
+-}
+data PCurrentTime (s :: S) = PCurrentTime
+    { lowerBound :: Term s PPOSIXTime
```

Comment:
I feel like we really don't need a record here since it's gonna be `pmatch`ed and users will pattern match it anyways. Happy to keep it here if there is a good reason I'm unaware of.

Reply:
My guess is record labels.